services/121-service/src/migration/1724754547895-modify-safaricom-requests-table.ts

```diff
@@ -0,0 +1,66 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class ModifySafaricomRequestsTable1724754547895
+  implements MigrationInterface
+{
+  name = 'ModifySafaricomRequestsTable1724754547895';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    // Step 1: Drop the columns that are not needed
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "initiatorName"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "securityCredential"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "commandID"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "amount"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "partyA"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "partyB"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "remarks"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "queueTimeOutURL"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "resultURL"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "occassion"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "status"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "requestResult"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" DROP COLUMN "paymentResult"`,
+    );
+
+    // Step 2: Rename the necessary columns
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" RENAME COLUMN "conversationID" TO "mpesaConversationId"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "121-service"."safaricom_request" RENAME COLUMN "transactionId" TO "mpesaTransactionId"`,
+    );
```

Comment:
Also rename column originatorConversationID to originatorConversationId (looks like a typo in the code and the diagram).

Reply:
Updated!